internal/config/image/builder.go

```diff
@@ -100,3 +110,33 @@ func WithMounts(mounts []specs.Mount) Option {
 		return nil
 	}
 }
+
+// WithPrivileged sets whether an image is privileged or not.
+func WithPrivileged(isPrivileged bool) Option {
+	return func(b *builder) error {
+		b.isPrivileged = isPrivileged
+		return nil
+	}
+}
+
+// WithAdditionalVisibleDevicesEnvVars sets the visible devices environment variables.
+// If the resource is a single string, it is split by comma and the resulting
+// strings are added to the visible devices environment variables.
+func WithAdditionalVisibleDevicesEnvVars(visibleDevicesEnvVars ...string) Option {
```

Comment:
The more I think about it, the more I think we should make this a simple setter and handle the specifics for the SWARM envvars in the `hookConfig` type. That is to say, we undelete `getSwarmResourceEnvvars()` and then just pass the outpuit of that into `WithAdditionalVisibleDevicesEnvVars`.